### PR TITLE
v1.10.0a9: 195 move sector loadersdynamic back to sector generators

### PR DESCRIPTION
**Pull request by mindyls**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#195

These PRs should all be ready to review/approve with these updates:
https://github.com/NRLMMD-GEOIPS/geoips/pull/196
https://github.com/NRLMMD-GEOIPS/geoips_clavrx/pull/6
https://github.com/NRLMMD-GEOIPS/recenter_tc/pull/18

# Testing Instructions
$GEOIPS/tests/test_full_install.sh

# Summary
Version 1.10.0a8 (2023-04-27)
*****************************

* Move sector_loaders.dynamic back to sector_generators

  * Originally was area_def_generators

Breaking Changes
================

Move sector_loaders.dynamic to sector_generators
------------------------------------------------

*From NRLMMD-GEOIPS/geoips#195: 2023-04-26, sector_generators*

These were originally called "area_def_generators", but we moved to
sector_loaders.dynamic originally to generalize for multiple use cases.
Since determined they really are generating area definitions, but
should be generalized to "sector_generators" in case we ever
implement a non-area_def based sector back end.

Note I did not update all of the YAML plugins - they will be updated in
a later PR.

::

  modified: sector_utils/tc_tracks.py
  modified: sector_utils/utils.py
  renamed: geoips/plugins/modules/sector_loaders/dynamic/__init__.py -> geoips/plugins/modules/sector_generators/__init__.py
  renamed: geoips/plugins/modules/sector_loaders/dynamic/clat_clon_resolution_shape.py -> geoips/plugins/modules/sector_generators/clat_clon_resolution_shape.py

# Output

geoips check_code (expected) and geoips_clavrx check_code failed (I should double check that one)!  Everything else passed!

Package: geoips_full
Total run time: 1835 seconds
Number data types run: 35
Number data types failed: 2

### CODE REVIEW

**Review by cpcamacho — APPROVED** [review]
The motivation and explanation for moving these modules back to sector_generators looks good to me!

**Review by jsolbrig — DISMISSED** [review]

**Comment by mindyls** [comment]
@jsolbrig @cpcamacho As discussed, updating this with both sector_spec_generators and sector_metadata_generators.  Will update the module-based plugin locations, and update the code accordingly.  Schema / YAML will be addressed in #190

File: geoips/plugins/modules/sector_metadata_generators/tc_sector_file_parser.py [hunk 1]
```diff
@@ -24,8 +23,12 @@
 
 LOG = logging.getLogger(__name__)
 
+interface = "sector_metadata_parser"
+family = "tc"
```
Comment (jsolbrig) [inline geoips/plugins/modules/sector_metadata_generators/tc_sector_file_parser.py:27]: Are we good with `family = "tc"` or would it be better to do `family = "tropical_cyclone"`? I've been using `tc`, too.

Comment (mindyls) [reply geoips/plugins/modules/sector_metadata_generators/tc_sector_file_parser.py:27]: We could actually do tropical_cyclone ... let's change it in a different PR though

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**